### PR TITLE
Add Tox and flake8 support

### PR DESCRIPTION
**Pull request by Myoldmopar**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,20 @@
 language: python
+
```
Comment (Myoldmopar) [inline .travis.yml]: Just cleaning it

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 python:
   - "3.6"
 
 cache: pip
 
+env:
+  matrix:
+    - TOX_ENV=python
+    - TOX_ENV=flake8
+
```
Comment (Myoldmopar) [inline .travis.yml]: Here's where we define each new environment for Travis to run.  For now, this is really just setting the value of an environment variable.

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
 install:
-  - "pip install -r requirements.txt"
-  - pip install coveralls
+  - pip install --upgrade pip
+  - pip install tox coveralls
 
 script:
-  - coverage run setup.py test
+  - tox -e $TOX_ENV
```
Comment (Myoldmopar) [inline .travis.yml]: The actual testing is to call tox -e TESTNAME and let it run what it finds in the tox.ini file.

File: .travis.yml [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
 
-after_success:
-  coveralls
+after_success: test $TOX_ENV = "python" && coveralls
```
Comment (Myoldmopar) [inline .travis.yml]: We need to verify that we only push to coveralls after we run the "python" test environment.

**Comment by mitchute** [comment]
Ha! Didn't know Travis would pick up branches. That's nice! 👍

**Comment by Myoldmopar** [comment]
Hahaha OMG so many flake8 things to fix :+1: 

Actually there are a bunch of them all reporting the same line, so they'll drop like flies

**Comment by Myoldmopar** [comment]
And I should say well done on having this set up nicely already.  A quick:

```
pyenv install 3.6.0     # install a 3.6.0 interpreter on my machine
pyenv virtualenv 3.6.0 OpenGHX    # create a virtual env inside my new 3.6.0 interpreter called OpenGHX
pyenv activate OpenGHX   # activate this new virtual env so I can install packages into it
pip install -r requirements.txt    # install your dependencies
python setup.py test    # run your tests
```

worked just fine :+1:

**Comment by mitchute** [comment]
Well, thanks for the help. We'll have to discuss the pyenv a little more next time. It would be good practice to learn more about it and use it.

**Comment by Myoldmopar** [comment]
I turned off flake8 support just by commenting one line.  That way this could get merged at your leisure and then you can turn that line back on and fix those issues whenever you want later.

**Comment by mitchute** [comment]
I ran autopep8 on everything. I wonder what it would look like now... Might turn it on and see.

**Comment by Myoldmopar** [comment]
Sounds like a tomorrow task :sleepy:

**Comment by mitchute** [comment]
Fo sho. Thanks!

**Pull request merged**